### ISSUE TITLE
Events created due to expired TTL do not increment occurences field

### ISSUE THREAD

**Issue by roganartu** [sensu/sensu-go#3118]
Exceeding the TTL set for a check correctly creates a new event and appends the current timestamp and TTL expired event status to history, however it does not increase the `event.check.occurrences` field.

## Expected Behavior
TTL events should increment the `event.check.occurrences` and `event.check.occurences_watermark` fields.

## Current Behavior
TTL events carry forward the `event.check.occurrences` and `event.check.occurrences_watermark` fields from the stale event that triggered them.

## Possible Solution
Dead checks are handled by the following snippet:
https://github.com/sensu/sensu-go/blob/88f30484eb2f68eeb151e53baaf68bd2ac095ac0/backend/eventd/eventd.go#L345-L359
which calls out to
https://github.com/sensu/sensu-go/blob/532133dbdf2fb5d1b5e807ff7495a54a9e7584b2/backend/store/etcd/event_store.go#L206-L289

I'm unsure why this isn't working as expected, afaict all code paths _should_ successfully call `updateOccurrences` with the new event. The only thing I can think is that `event.Check.MergeWith` might be erroneously called twice on the new event, but that definition is here:
https://github.com/sensu/sensu-go/blob/613f31c430c7d2b8d00de82e8815756f68f118c9/api/core/v2/check.go#L216-L234
and if this was being called twice I'd expect to see duplicated `history` entries which we don't see here.

## Steps to Reproduce (for bugs)
1. Submit the following event to Sensu
```
{
  "entity": {
    "entity_class": "proxy",
    "system": {
      "hostname": "teting"
    },
    "subscriptions": [],
    "deregister": false,
    "user": "sensu",
    "metadata": {
      "name": "testing",
      "namespace": "default"
    }
  },
  "check": {
    "handlers": [
      "nopage",
      "audit"
    ],
    "publish": false,
    "proxy_entity_name": "testing",
    "ttl": 30,
    "status": 0,
    "metadata": {
      "name": "test_occurrences",
      "namespace": "default"
    }
  },
  "metadata": {
    "namespace": "default"
  }
}
```
2. Wait 30s for the TTL to expire
3. Inspect the output of `sensuctl event info`and see that the `event.check.history` field contains multiple entries, but `event.check.occurrences` remains at `1`.
4. Repeat 2-3, noting that `event.check.history` continues to get new events every time the TTL expires, but `event.check.occurrences` never increments.

Example from our elasticsearch index which is fed by an unfiltered `audit` handler:
![Screenshot from 2019-07-05 14-54-30](https://user-images.githubusercontent.com/1133705/60740810-11aacc80-9f35-11e9-8dee-9a09065c1c96.png)
You can see in the image the initial event (created by POSTing to the agent endpoint, to avoid having to authenticate), and then the two subsequent TTL expiry events that were created. The status doesn't change, and you can see there are multiple events in the history, but `event.check.occurrences` remains at 1. If I was to POST the event to the agent multiple times in quick succession you can instead see that `event.check.occurrences` carries forward the previous value:
![Screenshot from 2019-07-05 15-00-06](https://user-images.githubusercontent.com/1133705/60740943-a57c9880-9f35-11e9-8a08-eee3b5d7ae38.png)

## Context
We use a custom label (`check_attempts`) in our check config which is checked by our custom filters. This label is used to enable our checks to require multiple consecutive failures before alerting (similar to the `max_check_attempts` check config option in Nagios, and presumably other tools).

This works fine for regular check events, however I noticed that none of our dead checks ever passed through the filters. On closer inspection, I noticed it was because they all had unusually low `event.check.occurrences` values considering they'd be failing for days.

## Your Environment
* Sensu version used (sensuctl, sensu-backend, and/or sensu-agent):
```
sensu-backend version 5.10.2#13bafd5, build 13bafd56ce60cb8c91838bfa4e65c6ff512c3d0d, built 2019-06-27T17:47:48Z
sensu-agent version 5.10.2#13bafd5, build 13bafd56ce60cb8c91838bfa4e65c6ff512c3d0d, built 2019-06-27T17:48:41Z
```
* Operating System and version (e.g. Ubuntu 14.04): Debian Stretch

**Comment by roganartu** [comment]
I'm not sure if this was an intended fix bundled in 5.11, but I upgraded to that yesterday and it appears to have resolved this issue.

An example of occurrences for a given check incrementing correctly:
![Screenshot from 2019-07-12 13-53-37](https://user-images.githubusercontent.com/1133705/61148359-93a67280-a4ac-11e9-87c6-d2be0e76951d.png)

**Comment by palourde** [comment]
@roganartu Sorry for the late response, I'm glad it appears to be fixed now!

I don't see anything specific in the 5.11.0 release that would directly fix this bug but it's possible it was a symptom of an another bug that got fixed.

I'm going to close this issue but please let us know if you still have this problem.
